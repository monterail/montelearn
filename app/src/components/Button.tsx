import React, { FC } from "react";

type Props = {
  className?: string;
  onClick?: VoidFunction;
};

const Button: FC<Props> = ({ children, className = "", onClick }) => (
  <button
    type="button"
    className={`
      flex font-medium text-white
      bg-red-monterail rounded-full font-roboto-mono
      ${className || ""}
    `}
    onClick={onClick}
  >
    {children}
  </button>
);

export default Button;
