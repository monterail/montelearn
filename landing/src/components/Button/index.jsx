import { memo } from "react";

import { COLOR_GREEN, COLOR_WHITE } from "@project/core/lib/const/theming";
import { hexToRGBA, rem } from "@project/core/lib/utils/theming";

function Button({ css = {}, palette, size, ...props }) {
  return (
    <button
      type="submit"
      css={{
        MozAppearance: "none",
        WebkitAppearance: "none",
        backgroundColor: palette === "white" ? "transparent" : COLOR_GREEN,
        border: palette === "white" ? `1px solid ${COLOR_WHITE}` : "none",
        borderRadius: rem(100),
        color: COLOR_WHITE,
        cursor: "pointer",
        flexShrink: 0,
        fontFamily: "inherit",
        fontSize: size === "large" ? rem(20) : rem(14),
        margin: 0,
        minHeight: size === "large" ? rem(48) : rem(32),
        padding: rem(0, 20),
        ...css,

        ":hover": {
          backgroundColor:
            palette === "white" ? hexToRGBA(COLOR_WHITE, 0.2) : hexToRGBA(COLOR_GREEN, 0.75),
          ...css[":hover"],
        },
      }}
      {...props}
    />
  );
}

export default memo(Button);
