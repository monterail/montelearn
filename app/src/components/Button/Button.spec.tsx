import React from "react";
import { shallow } from "enzyme";

import Button from "./Button";

describe("Button", () => {
  describe("is defined/matches snapshot", () => {
    it("should be defined", () => {
      expect(Button).toBeDefined();
    });
    it("should render correctly", () => {
      const tree = shallow(<Button />);
      expect(tree).toMatchSnapshot();
    });
  });

  describe("className", () => {
    const customClass = "custom-class";
    it("should render Title without custom class", () => {
      const wrapper = shallow(<Button />);
      expect(wrapper.find("button").hasClass(customClass)).toBe(false);
    });
    it("should render Title with custom class", () => {
      const wrapper = shallow(<Button className={customClass} />);
      expect(wrapper.find("button").hasClass(customClass)).toBe(true);
    });
  });

  describe("onClick", () => {
    const mockOnClick = jest.fn();
    it("should call mock function when button is clicked", () => {
      const wrapper = shallow(<Button onClick={mockOnClick} />);
      wrapper.simulate("click");
      expect(mockOnClick).toHaveBeenCalled();
    });
  });

  describe("children", () => {
    const buttonText = "Button text";
    it("should render Button with custom text", () => {
      const wrapper = shallow(<Button>{buttonText}</Button>);
      expect(wrapper.find("button").text()).toMatch(buttonText);
    });
  });
});
