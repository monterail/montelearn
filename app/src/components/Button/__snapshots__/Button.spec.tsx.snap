// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button is defined/matches snapshot should render correctly 1`] = `ShallowWrapper {}`;
